vm/actor/src/builtin/miner/mod.rs

```diff
@@ -3584,31 +3517,40 @@ where
         proof: params.proof,
         randomness,
         sealed_cid: params.sealed_cid,
-        unsealed_cid: commd,
+        unsealed_cid: commds[0],
     })
 }
 
 /// Requests the storage market actor compute the unsealed sector CID from a sector's deals.
-fn request_unsealed_sector_cid<BS, RT>(
+fn request_unsealed_sector_cids<BS, RT>(
     rt: &mut RT,
-    sector_type: RegisteredSealProof,
-    deal_ids: &[DealID],
-) -> Result<Cid, ActorError>
+    data_commitment_inputs: &[SectorDataSpec],
+) -> Result<Vec<Cid>, ActorError>
 where
     BS: BlockStore,
     RT: Runtime<BS>,
 {
-    let ret = rt.send(
-        *STORAGE_MARKET_ACTOR_ADDR,
-        MarketMethod::ComputeDataCommitment as u64,
-        Serialized::serialize(ComputeDataCommitmentParamsRef {
-            deal_ids,
-            sector_type,
-        })?,
-        TokenAmount::zero(),
-    )?;
-    let unsealed_cid: Cid = ret.deserialize()?;
-    Ok(unsealed_cid)
+    if data_commitment_inputs.is_empty() {
+        return Ok(vec![]);
+    }
+    let ret: ComputeDataCommitmentReturn = rt
+        .send(
+            *STORAGE_MARKET_ACTOR_ADDR,
+            MarketMethod::ComputeDataCommitment as u64,
+            Serialized::serialize(ComputeDataCommitmentParamsRef {
+                inputs: &data_commitment_inputs,
+            })?,
+            TokenAmount::zero(),
+        )?
+        .deserialize()?;
```

Comment:
Do you think we could do something like `.expect("failed request for unsealed sector CIDs");` ? Since the following is kinda implicit in Rust:
https://github.com/filecoin-project/specs-actors/pull/1381/files#diff-d807293f55563b410c2d07719d2f38727016f41c6a17df8635500abc4070f74cR939-R943

Reply:
No we cant. We need to return the proper exit codes and stuff. Should avoid panicing in the actors code.

Reply:
Oh, that's right. Good point, thanks!